kong/init.lua

```diff
@@ -147,6 +151,172 @@ local function load_plugins(kong_conf, dao)
   return sorted_plugins
 end
 
+
+local function mock_plugins(ctx, phase)
+  local mock    = ctx.mock     or {}
+  local plugins = mock.plugins or {}
+  local phase   = phase        or ngx.get_phase()
+
+  local BasePlugin = require "kong.plugins.base_plugin"
+
+  for plugin, plugin_conf in plugins_iterator(singletons.loaded_plugins, phase == "rewrite" or phase == "access") do
+    if plugin.handler[phase] ~= BasePlugin[phase] then
+      table.insert(plugins, {
+        name        = plugin.handler._name,
+        version     = plugin.handler.VERSION,
+        priority    = plugin.handler.PRIORITY,
+        no_consumer = plugin.schema.no_consumer,
+        phase       = phase,
+        config      = plugin_conf,
+      })
+    end
+  end
+
+  mock.plugins = plugins
+  ctx.mock     = mock
+end
+
+
+local function mock_consumer(ctx)
+  local consumer_id
+
+  local args = arguments.load({ multipart = false })
```

Comment:
Ouch, can't we use the existing public API here? It seems like the changes in `arguments.lua` are trying to implement what we already have (or something similar to) as [kong.tools.public()](https://github.com/Kong/kong/blob/master/kong/tools/public.lua) (which will become part of the Public Lua API)